ci/upload-codecov.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+#
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+if [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then
+    echo "Code coverage disabled in pull requests, exit successfully" \
+         "without further action."
+    exit 0
+fi
+
+if [ "${BUILD_TYPE:-Release}" != "Coverage" ]; then
+    echo "Code coverage not enabled, exit successfully without further action."
+    exit 0
+fi
+
+if [ -z "${CODECOV_TOKEN+x}" ]; then
```

Comment:
The `+x` here is extra, no? If you want to test for empty, just test for empty.

Reply:
If CODECOV_TOKEN is not set then the expansion fails (due to the `set -eu`).  We either remove the `-u` and use `${VAR?}` everywhere else, or use `${VAR+x}` here.

Reply:
If you want a nice message when `VAR` is empty, don't you want to use `"${VAR:-}"`?

`${VAR+x}` returns:

* `$VAR` iff `$VAR` is set
* empty if `$VAR` is empty
* `x` iff `$VAR` is unset

which is not exactly what you want: you want the last two cases to both be caught by `-z` which means you want to use `${VAR:+x}` but `${VAR:-}` is much simpler since you just care about empty and don't need the "x", so just use that.

Look at the output of this script:

```bash
#!/bin/bash

readonly EMPTY=""
readonly SET=y

echo "unset: ${UNSET+x}"
echo "empty: ${EMPTY+x}"
echo "  set: ${SET+x}"

echo "unset: ${UNSET:+x}"
echo "empty: ${EMPTY:+x}"
echo "  set: ${SET:+x}"
```